### PR TITLE
[minirntuple] Move RError.hxx / cxx to Foundation

### PR DESCRIPTION
**Pull request by oshadura**
First steps in direction of development RNTuple C API with minimal dependencies

### CODE REVIEW

**Comment by oshadura** [comment]
@pcanal @jblomer could you please take a look?

File: core/foundation/CMakeLists.txt [hunk 1]
```diff
@@ -23,24 +23,41 @@ set(Foundation_dict_headers
...
   src/TClassEdit.cxx
   src/TError.cxx
 )
 
-ROOT_OBJECT_LIBRARY(Foundation ${sources})
+if(CMAKE_CXX_STANDARD GREATER 11)
+  list(APPEND Foundation_dict_headers ROOT/RLogger.hxx)
+  list(APPEND FOUNDATION_SOURCES v7/src/RLogger.cxx)
+  set(FOUNDATION_V7_INC ${CMAKE_SOURCE_DIR}/core/foundation/v7/inc)
+  # TLogger.hxx may be used without root7 flag, but is placed in v7/inc/,
```
Comment (pcanal) [inline core/foundation/CMakeLists.txt:37]: Shouldn't we move TLogger.cxx out of the v7/inc directory since we moving it anyway?

**Review by jblomer — COMMENTED** [review]
Looks good to me!  I think @Axel-Naumann should have a quick look, too.

**Review by pcanal — CHANGES_REQUESTED** [review]
Possibly move TError.hxx out of v7 directory

**Comment by oshadura** [comment]
@pcanal do you mean RError.hxx? I was sure it requires C++14 (to be verified with @jblomer )

**Comment by oshadura** [comment]
I think @pcanal was talking about TLogger.hxx. I am checking!

**Comment by oshadura** [comment]
@pcanal TLogger.cxx is renamed in RLogger.cxx -> it was an old comment in cmake, I will cleanup it from everywhere https://github.com/root-project/root/commit/55d00faeba3c3b12c394e1d4d719e00901d08648

File: core/foundation/CMakeLists.txt (hunk 1 above)
Comment (oshadura) [reply core/foundation/CMakeLists.txt]: Good catch! Done!

**Comment by oshadura** [comment]
@pcanal as you see RLogger.hxx is using unique pointers, and sadly `make_unique` is a C++14 feature... I will revert back to the initial approach.

**Comment by pcanal** [comment]
@oshadura but we have a backport of make_unique ...

**Comment by oshadura** [comment]
@pcanal good point, I think everything is resolved now (file is moved and it is using ROOT/RMakeUnique.hxx). Can you please recheck?

**Review by pcanal — APPROVED** [review]

**Comment by oshadura** [comment]
@Axel-Naumann please can you have a quick look (suggestion of @jblomer) and I think it is ready to go!

**Pull request merged**